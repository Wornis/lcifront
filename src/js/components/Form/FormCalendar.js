import React from 'react';
import {DatePicker} from 'material-ui-pickers';
import AccessTimeIcon from '@material-ui/icons/AccessTime';
import format from 'date-fns/format';
import PropTypes from 'prop-types';

const FormCalendar = props => {
  const onChange = (newDate) =>
    props.onChangeDate(format(newDate, 'yyyy-MM-dd'));

  return (
    <div className="picker" style={{marginBottom: 15, width: 240}}>
      <AccessTimeIcon style={{marginTop: 20}} />
      <DatePicker
        disabled={props.disabled}
        id='formCalendarDatePicker'
        label="Date"
        value={props.selectedDate}
        style={{marginLeft: 10}}
        onChange={onChange.bind(this)}
        animateYearScrolling
        format="d MMM yyyy"
      />
    </div>
  );
};

FormCalendar.propTypes = {
  selectedDate: PropTypes.string.isRequired,
  onChangeDate: PropTypes.func.isRequired
};
export default FormCalendar;

