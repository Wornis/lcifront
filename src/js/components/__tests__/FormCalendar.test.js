import React from 'react';
import {createShallow} from '@material-ui/core/test-utils';
import FormCalendar from 'Components/Form/FormCalendar';
import format from "date-fns/format";

describe('Test FormCalendar component', () => {
    it('Renders correctly', () => {
        const wrapper = createShallow()(
            <FormCalendar
                onChangeDate={() => {}}
                selectedDate={format(new Date(2018, 3, 25), 'yyyy-MM-dd')}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('Successfully calling onChange on changing', () => {
        const doneChange = jest.fn();
        const wrapper = createShallow()(
            <FormCalendar
                onChangeDate={doneChange}
                selectedDate={format(new Date(), 'yyyy-MM-dd')}
            />
        );
        wrapper.find(`#formCalendarDatePicker`).simulate('change', new Date());
        expect(doneChange).toBeCalledWith(format(new Date(), 'yyyy-MM-dd'));
    });
});