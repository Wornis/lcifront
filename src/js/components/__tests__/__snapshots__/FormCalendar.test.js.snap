// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FormCalendar component Renders correctly 1`] = `
<div
  className="picker"
  style={
    Object {
      "marginBottom": 15,
      "width": 240,
    }
  }
>
  <pure(AccessTimeIcon)
    style={
      Object {
        "marginTop": 20,
      }
    }
  />
  <ForwardRef
    animateYearScrolling={true}
    format="d MMM yyyy"
    id="formCalendarDatePicker"
    label="Date"
    onChange={[Function]}
    style={
      Object {
        "marginLeft": 10,
      }
    }
    value="2018-04-25"
  />
</div>
`;
